msal/src/androidTest/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationConfigurationFactoryTest.kt

```diff
@@ -0,0 +1,108 @@
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.nativeauth
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.microsoft.identity.client.AndroidTestUtil
+import com.microsoft.identity.client.exception.MsalException
+import com.microsoft.identity.common.java.nativeauth.BuildValues
+import com.microsoft.identity.common.java.net.HttpUrlConnectionFactory
+import com.microsoft.identity.msal.test.R
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@ExperimentalCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class NativeAuthPublicClientApplicationConfigurationFactoryTest {
+    private lateinit var context: Context
+    private var cachedUseMockApi: Boolean = false
+
+    @Before
+    fun setup() {
+        System.setProperty(
+            "dexmaker.dexcache",
+            InstrumentationRegistry
+                .getInstrumentation()
+                .targetContext
+                .cacheDir
+                .path
+        )
+
+        System.setProperty(
+            "org.mockito.android.target",
+            ApplicationProvider
+                .getApplicationContext<Context>()
+                .cacheDir
+                .path
+        )
+
+        context = InstrumentationRegistry.getInstrumentation().context.applicationContext
+        cachedUseMockApi = BuildValues.shouldUseMockApiForNativeAuth()
+    }
+
+    @After
+    fun tearDown() {
+        BuildValues.setUseMockApiForNativeAuth(cachedUseMockApi)
+        HttpUrlConnectionFactory.clearMockedConnectionQueue()
+        AndroidTestUtil.removeAllTokens(context)
+    }
+
+    @Test
+    fun testNativeAuthorityConfigMockApiTrue() {
+        runBlocking {
+            try {
+                var config = NativeAuthPublicClientApplicationConfigurationFactory.initializeNativeAuthConfiguration(
+                    context, R.raw.test_native_auth_use_mock_api_true)
+
+                Assert.assertEquals(true, config.useMockAuthority)
```

Comment:
```suggestion
                Assert.assertTrue(config.useMockAuthority)
```

Reply:
useMockAuthority is nullable so I have kept assertEquals